### ISSUE TITLE
Splatoon 3 - All issues I encountered (Windows 10, Nvidia, Vulkan)

### ISSUE THREAD

**Issue by Lt-knb** [yuzu-emu/yuzu#8877]
Ryzen 7 2700
RAM 16 GB
Gtx 1070
Windows 10 1909
Nvidia driver game ready 512.15
Yuzu EA 2935

Splatoon 3 is (almost) perfectly playable day one on Windows 10 with an Nvidia GPU (at least on my specs), with just one major letdown in the form of massive loading times (2 to 10 minutes), when first building the shader cache.

My settings:
Network Interface: None
API: Vulkan (OpenGl causes major graphical issues atm)
Resolution: 2x (if the special boost issue (from Splatoon 2) makes you feel like a cheater, set this to 1x)
Window adapting filter: Bicubic
Accuracy Level: High (may fix some particles)
Asynchronous shader building: on (I don't know if disabling it makes a difference)

General issues:

-The shader cache will be affected by the long loading times you had when playing the first time.
As a consequence, once it's complete, it could take more than 15 minutes to build it, before the game boots.

-The game seemingly froze twice during my playthrough. The first time in a normal level, the second time in the final phase of the final boss. It's very rare.
Fixed by... waiting for that nasty shader to build.
It's not stuck forever, even if it looks so.
May be the same issue as the big loading times. If that's the case, it will only happen the first time.
Also, DON'T reset Yuzu during the final phase of single-player. You are forced to "binge" it, because it only saves at the very end.

-It only crashed twice. Random.
May be a shader cache issue.

-Some Splat-Switches activate themselves, or are activated just by aiming at them. Random. May render single-player either easier or unbeatable.
Fixed by an in-game level reset (or multiple ones).

-Level 5 of the first site of Alterna sometimes "beats itself" upon entering the kettle, for no apparent reason.
NPCs talking way earlier are a clear sign this is the case.
Fixed by either a level reset or a Yuzu reset.
It may be a shader cache issue. In that case, it will only happen in the first playthrough.

-Sometimes, you are extremely fast in swim form for a couple of secs, and it can toss you out of existence. Rare.
May render single-player either easier or more frustrating.
No fix.

-Temporary lag (game speed. 52 to 57 fps) whenever going to Alterna from a different location .
It also happens in some levels.
It eventually goes away, but I can't figure out when or why.

-#7855 Specials charge faster depending on Resolution Scaling. Happens in this game as well, and affects single-player.
Fixed by setting Resolution Scaling to 1x.

-Sunken Scroll 5 and 15 have glitched graphics.
No fix atm.

-Crackling sound when booting directly to the plaza. Lasts a couple secs.
It also happens whenever the rocket "rumbles".
Minor. No fix.

-The screen flickers for 2 secs in the plaza, whenever returning from single-player.
Almost imperceptible. No fix.

-The floor sometimes behaves as if there's enemy ink, but it's minor.
No fix.

Shader cache issues (when first building the shader cache, the following happens. Subseguent playthroughs are unaffected once the shader cache is complete):

-Massive loading times when entering a new area (2-10 minutes), including booting and the tutorial.
Only happens the first time you go there.
Later levels of single-player will load faster, but still not ideal. The last phase and boss-fight will go back to slow.
For some reason, entering an area from a different location will also load slow and build shaders again. Example: You went to Alterna from the plaza (and won't load slow the next time), but it will be slow again if you enter it from a level instead (as far as I know, it only happens the very first time, and other levels won't be affected).
On the other hand, a level reset will always be slow the first time, in every level (not entirely sure though).
Only reset a level if absolutely necessary.

-The tutorial may have major graphical issues.
Fixed by a Yuzu reset.

-Small parts of a level and/or objects may behave as if they are un-inkable (not often, but not rare either) ; some may be invisible (rare); textures may be glitched (very rare); and there could be vertex explosions (only happened once).
In my playthrough, it never made the game unbeatable except for one instance.
A Yuzu reset fixes all of these.

**Comment by Sciroccogti** [comment]
I'm using Ubuntu22.04, Nvidia RTX2070, RAM16GB.

In my case, OpenGL is the better choice, since the display of Vulkan is quite weird.

And 16GB ram seems to be a bottleneck for Splatoon3 to simulate well on my machine. Splatoon3 takes nearly 96% of my RAM, and BotW only takes 60%.

**Comment by noneistaken** [comment]
Also entering The Shoal (local play) gets you error code 2203-0023.
![2203-0023](https://user-images.githubusercontent.com/90960821/189579803-7ddb8609-b5ad-4cd7-b13a-7a5a4567398f.png)

**Comment by FoxivanFable** [comment]
I am currently at the part where you first enter alterna and have to walk up to the squid sisters, however my game decided that the snow is enemy ink and i cant make it up any of the ledges, ALSO i get an upgrade point whenever i ink anything,

**Comment by Lt-knb** [comment]
Getting upgrade points by inking the ground **is** the expected behavior, but resolution scaling affects it the same way as the special charge. If you are getting too many points, disable resolution scaling.

I didn't encounter that "snow is enemy ink" issue, especially if it affects the whole ground. Does a Yuzu reset not fix it?

**Comment by charlie-gray** [comment]
Steam Deck user here. Not sure if I have much to offer, but my experience is as follows.

When using Vulkan, I am unable to get past the opening screen before the game freezes indefinitely. I've only been able to get the game to function by using OpenGL with GLSL and the other settings above. The stuttering is still an absolute nightmare though.

**Comment by Lt-knb** [comment]
Did you wait for up to 10 minutes, or did you just _assume_ it froze forever?

**Comment by FoxivanFable** [comment]
> Getting upgrade points by inking the ground **is** the expected behavior, but resolution scaling affects it the same way as the special charge. If you are getting too many points, disable resolution scaling.
> 
> I didn't encounter that "snow is enemy ink" issue, especially if it affects the whole ground. Does a Yuzu reset not fix it?

is a Yuzu reset just closing and opening the application? or is it something more, i can post an image / video once i get home if you want

**Comment by Lt-knb** [comment]
Yes, that's enough.
Also, a video would be appreciated

**Comment by FoxivanFable** [comment]
https://user-images.githubusercontent.com/112157418/190077677-3025271a-17a1-4468-bd29-8da090feb78e.mp4

this seems to just be a think with vulkan

**Comment by Lt-knb** [comment]
Oh wow, that's not good. So, a reset doesn't fix it?
What are your settings and specs?

**Comment by SignorUpB** [comment]
could you publish the shader cache?

**Comment by Lt-knb** [comment]
I believe that's considered piracy here. It's encouraged to build your own because of this, and the fact "someone else's" may not be compatible and/or mess up something if I remember correctly

**Comment by vmartin123** [comment]
ink issue in snow still persists, any solution??

**Comment by Lt-knb** [comment]
Specs, settings, and Nvidia driver version?

**Comment by 0x384c0** [comment]
Switching to OpenGL solves ink issue in Alterna and long loading times. But Splatsville becomes too bright.

**Comment by Lt-knb** [comment]
Not only that: It also causes all stickers/graffiti to disappear (YFC part 1 regression), and shading looks flat

**Comment by vmartinmajorkeytech** [comment]
> Switching to OpenGL solves ink issue in Alterna and long loading times. But Splatsville becomes too bright.

Using OpenGL also has A LOT of visual glitches

**Comment by Lt-knb** [comment]
I confirm

**Comment by shayaansiddiqui** [comment]
> I believe that's considered piracy here. It's encouraged to build your own because of this, and the fact "someone else's" may not be compatible and/or mess up something if I remember correctly

I dont think its pirating because the Shader is generated. But you're right about it not working properly.

**Comment by Lt-knb** [comment]
It is according to the Discord's rules. No idea why though. Maybe it's because it contains game data, but then, the VG Resource should be illegal, and it's not, apparently.
It's so confusing.
Maybe it's one of those "gray areas"...

**Comment by G-Spawn** [comment]
Yes. Please avoid sharing your shaders for the game due to legality issues. Stick to ones you manually build.

**Comment by Inui-senpai** [comment]
I've tested this game on my PC (Ryzen 5 5600X, Radeon RX 580, 32 GB RAM (it had 16 GB RAM before, and emulator crashed everything on loading screen), Windows 10 22H2) and my laptop (Core i7-7700HQ, GeForce GTX 1050, 32 GB RAM, Windows 11 22H2). Both devices, when I launch game with OpenGL enabled, shows over-lit locations after 2 location changes. And when I launch game with Vulkan enabled, on PC all graphics (except the UI) transformed into 50 shades of grey (literally), but it looks normal on laptop, although I observe a periodically character blinking.
And yep, I've observed an ink issue on both renderers, but I already bought some upgrades that make this problem less annoying, so it's not that critical to me.

**Comment by titoudoc** [comment]
On latest EA build 3947 and latest Mainline build 1599, Yuzu get stuck at loading shader screen unless I delete all shader cache. If I delete all shader cache, game load and I can play (still long loading time of course).
I also got some brightness issue but reload the level correct the issue most of the time.
